src/Xml/Decode.elm

```diff
@@ -92,19 +101,59 @@ parseSlice first firstClose trimmed =
 
         closeTag =
             "</" ++ tagName ++ ">"
+
+        openTag =
+            "<" ++ Regex.escape tagName ++ "[\\s\\n]*[^/]*?>" 
+            |> Regex.regex
+
+        openTags =
+            Regex.find Regex.All openTag afterOpen
+                |> List.map (.index >> (,) OpenTag)
+
+        closeTags =
+            String.indexes closeTag trimmed
+                |> List.map ((,) CloseTag)
+
+        allTags =
+            openTags
+                ++ closeTags
+                |> List.sortBy second
+
+        reduce list =
+            case list of
+                ( OpenTag, i1 ) :: ( CloseTag, i2 ) :: rest ->
+                    reduce rest
+
+                ( OpenTag, i1 ) :: rest ->
+                    case reduce rest of
+                        ( CloseTag, i2 ) :: rest ->
+                            reduce rest
+
+                        rest ->
+                            rest
+
+                _ ->
+                    list
     in
-        case String.indexes closeTag trimmed of
-            [] ->
+        case List.head <| reduce allTags of
+            Nothing ->
                 if String.startsWith "?" tagName then
                     Ok ( DocType tagName props, firstClose + 1 )
-                else if (String.contains "/>" trimmed) then
-                    Ok ( Tag tagName props (Object []), firstClose + 1 )
+                else if String.endsWith "/" beforeClose then
+                    let tag = if String.endsWith "/" tagName then
```

Comment:
this should be on a newline

Reply:
Sorry. I just started using elm-format. Was naughty about such things in the past. Will fix. By saving in my editor. :)